samcli/lib/sync/flows/rest_api_sync_flow.py

```diff
@@ -48,15 +51,156 @@ def __init__(
+                        if rest_api_id == self._api_identifier:
+                            stages.add(cast(str, stage_dict.get("Properties").get("StageName")))  # type: ignore
+
+        return stages
+
+    def _update_stages(self, stages: Set[str], deployment_id: Optional[str]) -> Set[str]:
+        """
+        Update all the relevant stages
+
+        Parameters
+        ----------
+        stages: Set[str]
+            The set of stage names to be updated
+        deployment_id: Optional[str]
+            The newly created deployment ID to be used in the stages
+        Returns
+        ----------
+        Set[str]: A set of previous deployment IDs to be cleaned up
+        """
+        prev_dep_ids = set()
+        for stage in stages:
+            # Collects previous deployment IDs to clean up
+            response_get = cast(Dict, self._api_client.get_stage(restApiId=self._api_physical_id, stageName=stage))
+            prev_dep_id = response_get.get("deploymentId")
+            if prev_dep_id:
+                prev_dep_ids.add(cast(str, prev_dep_id))
+
+            # Updates the stage with newest deployment
+            LOG.debug("%sTrying to update the stage %s through client", self.log_prefix, stage)
+            response_upd = cast(
+                Dict,
+                self._api_client.update_stage(
+                    restApiId=self._api_physical_id,
+                    stageName=stage,
+                    patchOperations=[{"op": "replace", "path": "/deploymentId", "value": deployment_id}],
+                ),
+            )
+            LOG.debug("%sUpdate Stage Result: %s", self.log_prefix, response_upd)
+
+            # Flushes the cache so that the changes are calleable
+            self._api_client.flush_stage_cache(restApiId=self._api_physical_id, stageName=stage)
+            self._api_client.flush_stage_authorizers_cache(restApiId=self._api_physical_id, stageName=stage)
+        return prev_dep_ids
+
+    def _delete_deployments(self, prev_deployment_ids: Set[str]) -> None:
+        """
+        Delete the previous deployment
+
+        Parameters
+        ----------
+        prev_deployment_ids: Set[str]
+            A set of previous deployment IDs to be cleaned up
+        """
+        for prev_dep_id in prev_deployment_ids:
+            LOG.debug("%sTrying to delete the previous deployment %s through client", self.log_prefix, prev_dep_id)
+            try:
+                response_del = cast(
+                    Dict, self._api_client.delete_deployment(restApiId=self._api_physical_id, deploymentId=prev_dep_id)
+                )
+                LOG.debug("%sDelete Deployment Result: %s", self.log_prefix, response_del)
+            except ClientError:
+                LOG.warning(
+                    Colored().yellow(
+                        "Delete deployment for %s failed, it may be due to the it being used by another stage. \
+please check the console if you want to delete it"
```

Comment:
May not be true, but should we suggest them to run infra sync?

Reply:
Deployments are not manageable in AWS console so they will have to delete the stage if they want to have that deployment removed, I can update the text to be more descriptive. But infra sync doesn't do the job either. We don't need to guarantee a deployment clean up I would regard it as a good practice instead